app/api/v2/handlers/fact_api.py

```diff
@@ -32,27 +33,31 @@ def add_routes(self, app: web.Application):
     async def get_facts(self, request: web.Request):
         knowledge_svc_handle = self._api_manager.knowledge_svc
         fact_data = await self._api_manager.extract_data(request)
+        resp = []
         if fact_data:
             try:
                 store = await knowledge_svc_handle.get_facts(criteria=fact_data)
                 resp = await self._api_manager.verify_fact_integrity(store)
-                return web.json_response(dict(found=resp))
             except Exception as e:
                 self.log.warning(f'Encountered issue retrieving fact {fact_data} - {e}')
+                raise JsonHttpBadRequest(f'Unable to retrieve fact: {fact_data}')
```

Comment:
Do we also want to pass back the exception info? Same with get_relationships below